### PR TITLE
[RFC] Fix - Project Setup

### PR DESCRIPTION
**Pull request by Prakash-Kuikel**
`git fetch && git checkout feature/project-set-up`

#### Description
1. Added .git-ignore.
2. Added Rubocop configurations.
3. Added RSpec.
4. Added factory-bot.
5. Added test-prof.
6. Added shoulda-matchers.
7. Added pry-rails.
8. Added awesome-print.
9. Added and configured GraphQL

### CODE REVIEW

**Review by dorjikhandu21 — APPROVED** [review]
LGTM

File: app/graphql/types/mutation_type.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module Types
+  class MutationType < Types::BaseObject
+    # TODO: remove me
+    field :test_field, String, null: false,
```
Comment (rnzdoj) [inline app/graphql/types/mutation_type.rb:4]: remove this test field

File: app/graphql/types/query_type.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  class QueryType < Types::BaseObject
+    # Add `node(id: ID!) and `nodes(ids: [ID!]!)`
+    include GraphQL::Types::Relay::HasNodeField
+    include GraphQL::Types::Relay::HasNodesField
+
+    # Add root-level fields here.
+    # They will be entry points for queries on your schema.
+
+    # TODO: remove me
+    field :test_field, String, null: false,
```
Comment (rnzdoj) [inline app/graphql/types/query_type.rb:11]: this field as well

File: Gemfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  gem 'debug', platforms: %i[mri mingw x64_mingw]
+  gem 'rubocop-rails'
+  gem 'rubocop-performance'
+  gem 'rubocop-rake'
+  gem 'rubocop-rspec'
+  gem 'awesome_print'
+  gem 'pry-rails'
+  gem 'rspec-rails'
+  gem 'factory_bot_rails'
+end
```
Comment (rnzdoj) [inline Gemfile:25]: user pry instead of debug, and pry-rails.

File: Gemfile [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+group :development do
+end
+
+group :test do
+  gem 'shoulda-matchers'
+  gem 'test-prof'
+end
```
Comment (rnzdoj) [inline Gemfile:33]: new line here

File: Gemfile [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
```
Comment (rnzdoj) [inline Gemfile:1]: didn't lint this file?

**Review by rnzdoj — CHANGES_REQUESTED** [review]

File: app/graphql/types/mutation_type.rb (hunk 1 above)
Comment (Prakash-Kuikel) [reply app/graphql/types/mutation_type.rb:4]: Done

File: app/graphql/types/query_type.rb (hunk 2 above)
Comment (Prakash-Kuikel) [reply app/graphql/types/query_type.rb:11]: Done

File: Gemfile (hunk 3 above)
Comment (Prakash-Kuikel) [reply Gemfile:25]: Done

File: Gemfile (hunk 4 above)
Comment (Prakash-Kuikel) [reply Gemfile:33]: Done

File: Gemfile (hunk 5 above)
Comment (Prakash-Kuikel) [reply Gemfile:1]: Done

**Review by rnzdoj — APPROVED** [review]

**Pull request merged**